onmt/inputters/text_dataset.py

```diff
@@ -36,13 +35,11 @@ def read(self, sequences, side, _dir=None):
             yield {side: seq, "indices": i}
 
 
-class TextDataset(DatasetBase):
-    @staticmethod
-    def sort_key(ex):
-        """Sort using the number of tokens in the sequence."""
-        if hasattr(ex, "tgt"):
-            return len(ex.src[0]), len(ex.tgt[0])
-        return len(ex.src[0])
+def text_sort_key(ex):
```

Comment:
The style of defining `sort_key` as a static method of a dataset class was to match torchtext, where I think the general idea is people would define a different dataset class for each dataset (classes with names like `IWSLT` and `WMT14`, for example). But for the way we're doing it, with just the single `Dataset` class, I agree that this makes sense.